src/Aspire.Hosting/LaunchProfileExtensions.cs

```diff
@@ -30,26 +30,37 @@ internal static class LaunchProfileExtensions
 
     internal static NamedLaunchProfile? GetEffectiveLaunchProfile(this ProjectResource projectResource, bool throwIfNotFound = false)
     {
-        string? launchProfileName = projectResource.SelectLaunchProfileName();
-        if (string.IsNullOrEmpty(launchProfileName))
+        try
         {
-            return null;
-        }
+            string? launchProfileName = projectResource.SelectLaunchProfileName();
+            if (string.IsNullOrEmpty(launchProfileName))
+            {
+                return null;
+            }
 
-        var profiles = projectResource.GetLaunchSettings()?.Profiles;
-        if (profiles is null)
-        {
-            return null;
-        }
+            var profiles = projectResource.GetLaunchSettings()?.Profiles;
+            if (profiles is null)
+            {
+                return null;
+            }
+
+            var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
+            if (!found && throwIfNotFound)
+            {
+                var message = string.Format(CultureInfo.InvariantCulture, Resources.LaunchSettingsFileDoesNotContainProfileExceptionMessage, launchProfileName);
+                throw new DistributedApplicationException(message);
+            }
 
-        var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
-        if (!found && throwIfNotFound)
+            return launchProfile is not null ? new (launchProfileName, launchProfile) : default;
+        }
+        catch (JsonException ex)
```

Comment:
Would it be better if this `try-catch` was moved closer to what was actually doing the Json parsing?

Specifically, if we moved it around lines 93-95 below:

```C#
        using var stream = File.OpenRead(launchSettingsFilePath);
        var settings = JsonSerializer.Deserialize(stream, LaunchSettingsSerializerContext.Default.LaunchSettings);
        return settings;
```

Then we wouldn't need to recreate the file path here, we would have it from the rest of the method.